### DISCUSSION TITLE
How do I customize the margin value

### DISCUSSION THREAD

**Discussion by laterdayi** [unocss/unocss#2167] [Q&A]
### Clear and concise description of the problem

How do I customize the margin value

The default mr mt ml mb is 1em，How to set mr ml mt mb Custom 18px or 20px

### Suggested solution

How do I customize the margin value

### Alternative

How do I customize the margin value

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/unocss/unocss/blob/main/CONTRIBUTING.md).
- [X] Read the `README.md` of using the package.
- [X] Already used the [Interactive Docs](https://uno.antfu.me).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by laterdayi** [comment]
How to do it   margin & padding 

my:  margin : 20px 0 
mx:  margin : 0 20px
ml :  margin-left: 20px
mt:  margin-top: 20px

**Reply by jojojojojoj5564656465465** [reply]
`rules: [

    [/^z-(\d+)$/, ([, d]) => ({ 'z-index': `${d}` })],

    [/^h-(.+)$/, ([, a]) => {

      const numReg = /^((?![A-Za-z]).)*$/

      const h = numReg.test(a) ? `${a}%` : a

      return { height: h }

    }],

    [/^w-(.+)$/, ([, a]) => {

      const numReg = /^((?![A-Za-z]).)*$/

      const w = numReg.test(a) ? `${a}%` : a

      return { width: w }

    }],

    [/^p-(\w+)-(\w+)?-?(\w+)?-?(\w+)?$/, ([, t, r, b, l]) => {

      const effectiveArr = [t, r, b, l].filter(e => e)

      const numReg = /^((?![A-Za-z]).)*$/

      const paddingList = effectiveArr.map(e => numReg.test(e) ? `${e}%` : e)

      return effectiveArr.length === 0 ? { padding: '0px' } : { padding: paddingList.join(' ') }

    }],

    [/^m-(\w+)-(\w+)?-?(\w+)?-?(\w+)?$/, ([, t, r, b, l]) => {

      const effectiveArr = [t, r, b, l].filter(e => e)

      const numReg = /^((?![A-Za-z]).)*$/

      const marginList = effectiveArr.map(e => numReg.test(e) ? `${e}%` : e)

      return effectiveArr.length === 0 ? { margin: '0px' } : { margin: marginList.join(' ') }

    }],

`

**Reply by jojojojojoj5564656465465** [reply]
https://pastebin.com/R9TdBaSj you have many exemple

**Comment by laterdayi** [comment]
@antfu   Looking forward to your help, how to override the default `mr mt ml mb pl pt `
`
